agent/consul/server_connect.go

```diff
@@ -219,21 +218,22 @@ func (s *Server) SignCertificate(csr *x509.CertificateRequest, spiffeID connect.
 
 	// Append any intermediates needed by this root.
 	for _, p := range caRoot.IntermediateCerts {
-		pem = strings.TrimSpace(pem) + "\n" + p
+		pem = ca.AddSingleNewline(pem) + p
 	}
 
 	// Append our local CA's intermediate if there is one.
 	inter, err := provider.ActiveIntermediate()
 	if err != nil {
 		return nil, err
 	}
+	inter = ca.AddSingleNewline(inter)
 	root, err := provider.ActiveRoot()
 	if err != nil {
 		return nil, err
 	}
-
+	root = ca.AddSingleNewline(root)
```

Comment:
Ahh, this is an interesting case. We actually want certs to have consistent trailing newlines otherwise the comparison fails. Good catch!

I wonder if this is a good reason to add the trailing newline in the vault provider.